sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -504,6 +504,9 @@ case class ArrayContains(left: Expression, right: Expression)
 
   override def dataType: DataType = BooleanType
 
+  @transient private lazy val ordering: Ordering[Any] =
+    TypeUtils.getInterpretedOrdering(right.dataType)
```

Comment:
Then in `checkInputDataTypes` we should check if there is `Ordering` for `right.dataType`. Otherwise for example MapType will throw a match error.

Reply:
yes, thanks